alibi/explainers/tests/test_ale.py

```diff
@@ -117,6 +117,11 @@ def test_explain(mock_ale_explainer, features, input_dim, batch_size, custom_gri
     if not custom_grid:
         for alev, featv in zip(exp.ale_values, exp.feature_values):
             assert alev.shape == (featv.shape[0], out_dim)
+    else:
+        for i, f in enumerate(grid_points.keys()):
+            # need to remove the first and last element just in case the feature values are extended
+            # with the min & max feature value
+            assert np.all(np.isin(exp.feature_values[i][1:-1], grid_points[f]))
```

Comment:
I'm not sure I follow the logic this test branch, the original branch was intended to check that `exp.ale_values` and `exp.feature_values` match in dimension so can be plotted, whereas this branch now considers the input `grid_points` rather than the output contained in `exp`?

Reply:
Perhaps we need 2 tests?
 - Check that `exp.ale_values` and `exp.feature_values` always match (so can be plotted)
 - In the case that `grid_points` is passed, check that the ultimate `exp.feature_values` match with the original `grid_points` (up to potentially adding the endpoints - however, would this not fail if there are gridpoints passed that don't contain any feature values?)